### PR TITLE
Add info about Gammapy contact points and gammapy-extra

### PR DESCRIPTION
**Pull request by cdeil**
This PR adds some info for users and developers on
- [x] Gammapy contact points, especially the recently added https://gammapy.slack.com/
- [x] https://github.com/gammapy/gammapy-extra

I don't have time to improve / finish the sample data loading stuff today ... this will have to wait for a future PR.

### CODE REVIEW

File: docs/datasets/index.rst [hunk 1]
```diff
@@ -23,6 +23,53 @@ The functions have a naming pattern (following the `sklearn.datasets` lead):
...
+
+gammapy-extra
+=============
+
+To keep the Gammapy code repository at https://github.com/gammapy/gammapy small and clean,
+we are putting sample data files and IPython notebooks in an extra repository
+at https://github.com/gammapy/gammapy-extra/ .
+
+To get the repository, ``git clone`` it to a location of your choosing using a git protocol of your choosing
+(try HTTPS or see the `Github clone URL help article`_ if you're not sure which you want).
```
Comment (bsipocz) [inline docs/datasets/index.rst]: this link doesn't work, change the ref name here or below

File: docs/datasets/index.rst [hunk 2]
```diff
@@ -23,6 +23,53 @@ The functions have a naming pattern (following the `sklearn.datasets` lead):
...
+The Gammapy tests, docs generator, examples and tutorials will access files from the ``gammapy-extra``
+repo using the `gammapy.datasets.gammapy_extra` object.
+
+For this to work, you have to set the ``GAMMAPY_EXTRA`` shell environment variable to point to that folder.
+We suggest you put this in you ``.bashrc`` or ``.profile``
+
+    export GAMMAPY_EXTRA=/path/on/your/machine/to/gammapy-extra
+
+After you've done this, open up a new terminal (or ``source .profile``) and check if ``gammapy-extra`` is found:
+
```
Comment (JonathanDHarris) [inline docs/datasets/index.rst]: This is still not dumbed down enough for me : P

I've added the environment variable:
```
jon@Treebeard:~/Pycharm/gammapy$ echo $GAMMAPY_EXTRA
/home/jon/Pycharm/gammapy-extra/gammapy-extra
```

but when I run my script

```
 File "/home/jon/Pycharm/gammapy/gammapy/datasets/core.py", line 244, in dir
    raise GammapyExtraNotFoundError(msg)
gammapy.datasets.core.GammapyExtraNotFoundError: The gammapy-extra repo is not available. You have to set the GAMMAPY_EXTRA environment variable to point to the location for it to be found.
````

Comment (cdeil) [inline docs/datasets/index.rst]: Is the `gammapy-extra` repo really checked out at `/home/jon/Pycharm/gammapy-extra/gammapy-extra`?

Maybe it's at ``/home/jon/Pycharm/gammapy-extra/`?

This is the check I put in:
https://github.com/gammapy/gammapy/blob/master/gammapy/datasets/core.py#L226

Do you have that file? At what location?

Comment (JonathanDHarris) [inline docs/datasets/index.rst]: You're correct, but...
```
jon@Treebeard:~/Pycharm/gammapy$ echo $GAMMAPY_EXTRA
/home/jon/Pycharm/gammapy-extra/
```
```
gammapy.datasets.core.GammapyExtraNotFoundError: The gammapy-extra repo is not available. You have to set the GAMMAPY_EXTRA environment variable to point to the location for it to be found.
```

The file is  there
```
jon@Treebeard:~/Pycharm/gammapy$ ls $GAMMAPY_EXTRA/logo/gammapy_logo.pdf
/home/jon/Pycharm/gammapy-extra/logo/gammapy_logo.pdf
```
For reference, this works
```
os.environ["GAMMAPY_EXTRA"] = "/home/jon/Pycharm/gammapy-extra"
```

Comment (cdeil) [inline docs/datasets/index.rst]: @JonathanDHarris - Looks like the `GAMMAPY_EXTRA` shell variable is not available from your Python process.
I'm not sure how to debug / fix this ... it depends on your system and how you start Python or IPython whether environment variables are available or not.

One thing I learned from this is that the error message should be improved.
It should first check if the `GAMMAPY_EXTRA` variable is available and if not explicitly say that this is the issue.
Only if it is available, should it check if the variable actually points to a folder where `gammapy-extra` is located, and if not, explicitly state that as an error.
These could be two different errors, e.g. add `GammapyExtraVariableNotSet`.

@JonathanDHarris - Do you agree that would be better? If yes, I could implement this change in this PR now.

Comment (cdeil) [inline docs/datasets/index.rst]: One thing you could try is to run Python this way:
```
GAMMAPY_EXTRA=/home/jon/Pycharm/gammapy-extra python myscript.py
```
Can you then see the environment variable from `myscript.py`?

File: docs/datasets/index.rst (hunk 1 above)
Comment (cdeil) [inline docs/datasets/index.rst]: Fixed.

**Comment by cdeil** [comment]
This build errored:
https://travis-ci.org/gammapy/gammapy/jobs/107430524#L1078
```
gammapy/astro/population/tests/test_spatial.py::TestMorphologyModels::test_fitter1D[model_class0-test_parameters0] <- ../../home/travis/miniconda/envs/test/lib/python3.5/site-packages/astropy/modeling/tests/test_models.py 
```
No idea what the issue is ... restarting the build now.

**Pull request merged**